plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/MonitorStatechartgenerator.xtend

```diff
@@ -0,0 +1,240 @@
+package hu.bme.mit.gamma.scenario.statechart.generator
+
+import hu.bme.mit.gamma.scenario.model.AlternativeCombinedFragment
+import hu.bme.mit.gamma.scenario.model.Delay
+import hu.bme.mit.gamma.scenario.model.InteractionDirection
+import hu.bme.mit.gamma.scenario.model.InteractionFragment
+import hu.bme.mit.gamma.scenario.model.LoopCombinedFragment
+import hu.bme.mit.gamma.scenario.model.ModalInteractionSet
+import hu.bme.mit.gamma.scenario.model.ModalityType
+import hu.bme.mit.gamma.scenario.model.NegatedModalInteraction
+import hu.bme.mit.gamma.scenario.model.OptionalCombinedFragment
+import hu.bme.mit.gamma.scenario.model.ScenarioDefinition
+import hu.bme.mit.gamma.statechart.contract.NotDefinedEventMode
+import hu.bme.mit.gamma.statechart.interface_.Component
+import hu.bme.mit.gamma.statechart.interface_.TimeUnit
+import hu.bme.mit.gamma.statechart.statechart.State
+import hu.bme.mit.gamma.statechart.statechart.StateNode
+import hu.bme.mit.gamma.statechart.statechart.TransitionPriority
+import java.math.BigInteger
+import java.util.List
+
+import static extension hu.bme.mit.gamma.statechart.derivedfeatures.StatechartModelDerivedFeatures.*
+
+class MonitorStatechartgenerator extends AbstractContractStatechartGeneration {
+
+	protected boolean skipNextinteraction = false
```

Comment:
`skipNextInteraction˙?

Reply:
In the case of an optional combined fragment, the interaction after the optional is processed also, thus, the next interaction should be skipped, to avoid double processing

Reply:
I meant you should use consistent naming conventions.